projects/hello_world/src/main.c

```diff
@@ -0,0 +1,46 @@
+#include "delay.h"  // For real-time delays
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for the wait function
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+  bool half_second;
+} Counters;
+
+void callback(SoftTimerId timer_id, void *context) {
+  Counters *counters = context;
+
+  if (counters->half_second == true) {
+    counters->counter_a++;
+    LOG_DEBUG("Counter A: %i\n", counters->counter_a);
+  } else {
+    counters->counter_a++;
+    counters->counter_b++;
+    LOG_DEBUG("Counter A: %i\n", counters->counter_a);
+    LOG_DEBUG("Counter B: %i\n", counters->counter_b);
+  }
+
+  counters->half_second = !counters->half_second;
+
+  soft_timer_start_millis(500, callback, counters, NULL);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+
+  Counters counters;
```

Comment:
we usually initialize an empty struct with `= { 0 };`, please use this for clarity. For reference, doing this initializes the memory of the struct to zero.

Reply:
will this work for the boolean being set to false? I understand why it would work for just numerical values.

Reply:
yes, a false boolean in C is equivalent to 0